POEApi.Transport/HttpTransport.cs

```diff
@@ -77,13 +80,30 @@ public bool Authenticate(string email, SecureString password, bool useSessionID)
                 }
 
                 credentialCookies.Add(new Cookie("POESESSID", unwrappedPassword, "/", ".pathofexile.com"));
-                using (var sessionIdLoginResponse = BuildHttpRequestAndGetResponse(HttpMethod.GET, loginURL))
+
+                try
                 {
-                    // If the response URI is the login URL, then the login action failed.
-                    if (sessionIdLoginResponse.ResponseUri.ToString() == loginURL)
-                        throw new LogonFailedException();
+                    TraditionalSessionIdLogin();
+                    return true;
+                }
+                catch (Exception ex)
+                {
+                    //Something that isn't a cloud flare exception occurred
+                    if (ex.Message.Contains("503") == false)
+                    {
+                        throw;
+                    }
+
+                    try
+                    {
+                        CloudFlareSessionIdLogin();
+                        return true;
+                    }
+                    catch

```

Comment:
Since we're just rethrowing any exceptions, this try/catch block around this is useless.